aws/table_aws_route53_vpc_association_authorization.go

```diff
@@ -0,0 +1,144 @@
+			Tags:    map[string]string{"service": "route53", "action": "ListVPCAssociationAuthorizations"},
+			KeyColumns: plugin.KeyColumnSlice{
+				{
+					Name:    "hosted_zone_id",
+					Require: plugin.Required,
+				},
+			},
+		},
+		Columns: awsGlobalRegionColumns([]*plugin.Column{
+			{
+				Name:        "hosted_zone_id",
+				Description: "The ID of the hosted zone for which you want a list of VPCs that can be associated with the hosted zone.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "vpc_id",
+				Description: "(Private hosted zones only) The ID of an Amazon VPC.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("VPCId"),
+			},
+			{
+				Name:        "vpc_region",
+				Description: "(Private hosted zones only) The region that an Amazon VPC was created in.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("VPCRegion"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("HostedZoneId", "VPCId"),
+			},
+			// no tags for this resource as per aws docs
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(authorizationAkas).Transform(transform.ToString),
+			},
+		}),
+	}
+}
+
+type VPCAssociationAuthorizationResult struct {
+	HostedZoneId *string
+	VPCId        *string
+	VPCRegion    *string
+}
+
+func authorizationAkas(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	result := d.HydrateItem.(*VPCAssociationAuthorizationResult)
+
+	// same format as terraform builtin IDs "HOSTED_ZONE_ID:VPC_ID"
+	// https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_vpc_association_authorization#import
+	akas := fmt.Sprintf("%s:%s", *result.HostedZoneId, *result.VPCId)
+	return akas, nil
+}
+
+// // LIST FUNCTION
+func listVPCAssociationAuthorization(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create session
+	svc, err := Route53Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_route53_zone_vpc_association_authorization.listVPCAssociationAuthorization", "client_error", err)
+		return nil, err
+	}
+
+	hostedZoneId := d.EqualsQualString("hosted_zone_id")
+
+	// Empty check
+	if hostedZoneId == "" {
+		return nil, nil
+	}
+
+	// https://docs.aws.amazon.com/Route53/latest/APIReference/API_ListVPCAssociationAuthorizations.html#API_ListVPCAssociationAuthorizations_RequestSyntax
+	// Max 50 per page as per aws api docs
+	maxItems := int32(50)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 1 {
+				maxItems = int32(1)
+			} else {
+				maxItems = int32(limit)
+			}
+		}
```

Comment:
Since Steampipe already handles situations where the limit value is less than 1. Could we update the logic as follows?

```suggestion
                if limit < maxItems {
			maxItems = int32(limit)
		}
```